web/client/components/data/query/DateField.jsx

```diff
@@ -44,39 +60,57 @@ class DateField extends React.Component {
         onUpdateField: () => {},
         onUpdateExceptionField: () => {}
     };
-
     render() {
+        // these values are already parsed by the enhancer
+        const startdate = this.props.fieldValue && this.props.fieldValue.startDate || null;
+        const enddate = this.props.fieldValue && this.props.fieldValue.endDate || null;
+
+        // needed to initialize the time parts to 00:00:00
+        const defaultCurrentDate = moment().startOf("day").toDate();
```

Comment:
Don't know why this is here (and in date filter) if is needed to make UTCDateTimePicker work, it should be inside it (enhancer, setting, for instance moment().startOf("day").toDate() as default value.

If has another reason, please explain better.

Reply:
This is needed to initialize the tiem part

Imagine the "date" case:
DateTimePicker component is initialized with a new Date based on that moment and it includes the time part.

then you select a date from the calendar but the time part will remain untouched.
with defaultCurrentDate it changes the time part of the initial value 

notice that it does that also for date-time case (we may want to not do it for that case)
i think we can put it in the enhancer enabled by a flag or by the type

Reply:
i think that the way we are using "defaultCurrentDate" is not needed in general,
but is needed only if you want to start with a 00:00 hh:ss which is not always the case

Reply:
@offtherailz I'll update the enhnacer with the defaultCurrentDate = moment().startOf("day").toDate();  at least for "date" format